osu.Game/Rulesets/Judgements/DrawableJudgement.cs

```diff
@@ -130,19 +130,24 @@ private void prepareDrawables()
             if (type == currentDrawableType)
                 return;
 
-            InternalChild = JudgementBody = new Container
+            // sub-classes might have added their own children that would be removed here if .InternalChild was used.
+            if (JudgementBody != null)
```

Comment:
not 100% sure on this (as opposed to having a `CreateBody()` body which also handles things like lighting) but let's go with it for now.

Reply:
Is this required at all? What's the purpose of remove-then-add? It should probably be documented if there's a specific reason for it.

As in, could this be just:
```csharp
if (JudgementBody == null)
    AddInternal(JudgementBody = new ...);
```

Reply:
If this is something that _has_ to happen, be it for ordering purposes, it may even be worthwhile to split this out into separate methods like `RemoveBody()` _and_ `CreateBody()`.

Reply:
See the other part of this PR, where hit lighting is added once, ever.

Reply:
The remove-then-add is keeping the existing logic, which is to update the `JudgementBody` to the new judgement type (see previous code, `InternalChild = `).